index/miner/types.go

```diff
@@ -0,0 +1,55 @@
+package miner
+
+import (
+	cbor "github.com/ipfs/go-ipld-cbor"
+	"time"
+)
+
+func init() {
+	cbor.RegisterCborType(Index{})
+	cbor.RegisterCborType(ChainIndex{})
+	cbor.RegisterCborType(Power{})
+	cbor.RegisterCborType(MetaIndex{})
+	cbor.RegisterCborType(Meta{})
+	cbor.RegisterCborType(Location{})
+}
+
+// Index contains on-chain and off-chain information about miners
+type Index struct {
+	Meta  MetaIndex
+	Chain ChainIndex
```

Comment:
Now the Index has separated metadata and on-chain information.
This because those different informations are mantained by two different subcomponents (`meta.go` and `onchain.go`), so the shouldn't use locks or care about concurrency.